app/routes/traces/post.js

```diff
@@ -0,0 +1,111 @@
+import Boom from '@hapi/boom';
+import Joi from '@hapi/joi';
+import db from '../../services/db';
+import logger from '../../services/logger';
+import traces from '../../models/traces';
+
+/**
+ * @api {post} /traces POST
+ * @apiGroup traces
+ * @apiDescription Upload a trace.
+ *
+ * @apiParam {object} [tracejson] TraceJSON object.
+ *
+ * @apiSuccess {object}   tracejson       TraceJSON object, with properties populated with id, timestamps and description.
+ *
+ * @apiError statusCode     The error code
+ * @apiError error          Error name
+ * @apiError message        Error message
+ * @apiErrorExample {json} Error Response:
+ *     HTTP/1.1 400 Bad Request
+ *     {
+ *      "statusCode": 400,
+ *      "error": "Bad Request",
+ *      "message": "Oops!"
+ *     }
+ */
+export default [
+  {
+    path: '/traces',
+    method: ['POST'],
+    options: {
+      auth: 'jwt',
+      validate: {
+        payload: Joi.object({
+          tracejson: Joi.object({
+            type: Joi.valid('Feature'),
+            properties: Joi.object({
+              description: Joi.string(),
+              timestamps: Joi.array()
+                .min(1)
```

Comment:
Should we require a `min` of at least two timestamps and two points? Since a trace with a single point does not make much sense?

Reply:
Changed.